src/Microsoft.CodeQuality.Analyzers.Exp/UnitTests/Usage/DisposeMethodsShouldCallBaseClassDisposeTests.cs

```diff
@@ -352,6 +352,252 @@ End Sub
 End Class");
         }
 
+        [Fact, WorkItem(1796, "https://github.com/dotnet/roslyn-analyzers/issues/1796")]
+        public void BaseDisposeAsyncCall_NoDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Threading.Tasks;
+
+class A : IDisposable
+{
+    public void Dispose() => DisposeAsync();
+
+    public virtual Task DisposeAsync()
+    {
+        return Task.CompletedTask;
+    }
+}
+
+class B : A
+{
+    public override Task DisposeAsync()
+    {
+        return base.DisposeAsync();
+    }
+}
+");
+
+            VerifyBasic(@"
+Imports System
+Imports System.Threading.Tasks
+
+Class A
+    Implements IDisposable
+
+    Public Sub Dispose() Implements IDisposable.Dispose
+        DisposeAsync()
+    End Sub
+
+    Public Overridable Function DisposeAsync() As Task
+        Return Task.CompletedTask
+    End Function
+End Class
+
+Class B
+    Inherits A
+
+    Public Overrides Function DisposeAsync() As Task
+        Return MyBase.DisposeAsync()
+    End Function
+End Class");
+        }
+
+        [Fact, WorkItem(1796, "https://github.com/dotnet/roslyn-analyzers/issues/1796")]
+        public void NoBaseDisposeAsyncCall_Diagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Threading.Tasks;
+
+class A : IDisposable
+{
+    public void Dispose() => DisposeAsync();
+
+    public virtual Task DisposeAsync()
+    {
+        return Task.CompletedTask;
+    }
+}
+
+class B : A
+{
+    public override Task DisposeAsync()
+    {
+        return Task.CompletedTask;
+    }
+}
+",
+            // Test0.cs(17,26): warning CA2215: Ensure that method 'Task B.DisposeAsync()' calls 'base.DisposeAsync()' in all possible control flow paths.
```

Comment:
does it care whether wait base.XXX or return base.XXX or is it fine as long as there is base.XXX ?

Reply:
For now we only validate that `base.DisposeAsync` is invoked, not whether it is awaited. This should help remove all the false positives at least.